File: python/字典(对象)/删除字典某值/pop.py
# 字典 pop() 方法删除字典给定键 key 所对应的值，返回值为被删除的值。key值必须给出。 否则，返回default值。

# pop(key[,default])

# 参数
    # key: 要删除的键值
    # default: 如果没有 key，返回 default 值

# 返回值 返回被删除的值。

Object = {
    "name": "Tom",
    "age": 18
}

print(Object.pop('name')) # Tom

print(Object) # {'age': 18}

print(Object.pop('a', 'nokey')) # nokey

File: javascript/数组/删除元素/pop.js
// 语法

// Array.pop()

// 说明

// pop() 方法将删除 array 的最后一个元素，把数组长度减 1，并且返回它删除的元素的值。
// 如果数组已经为空，则 pop() 不改变数组，并返回 ★undefined★ 值。


let a = [
    { name: '没有梦想' },
    { name: '何必留下' }
]

let b = a.pop();

console.log(b); // { name: '何必留下' }